com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -78,6 +78,85 @@ public function upgrade_1006() {
     return TRUE;
   }
 
+  /**
+   * Updates job role having wrong funder id with correct contact id
+   *
+   * @return bool
+   * @throws CiviCRM_API3_Exception
+   */
+  public function upgrade_1007() {
+    $contactDetails = $this->getContactsWithJobRole();
+    $jobRoles = civicrm_api3('HrJobRoles', 'get', [
+      'funder' => ['!=' => '|'],
+    ]);
+
+    foreach($jobRoles['values'] as $key => $value) {
+      $funders = explode('|', $value['funder']);
+      $funders = array_filter($funders);
+
+      list($funders, $allowUpdate) = $this->changeFunderToCorrectContactId($funders, $contactDetails);
+      if ($allowUpdate) {
+        civicrm_api3('HrJobRoles', 'create', [
+          'id' => $key,
+          'funder' => '|' . implode('|', $funders) . '|',
+        ]);
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Retrieves the correct contact for a funder and
+   * indicates update is required when found
+   *
+   * @param $funders
+   * @param $contactDetails
+   *
+   * @return array
+   */
+  private function changeFunderToCorrectContactId($funders, $contactDetails) {
+    $allowUpdate = FALSE;
+    foreach ($funders as $index => $funder) {
+      $jobRoleIds = array_column($contactDetails, 'jobrole_id');
+      $funderIndex = array_search($funder, $jobRoleIds);
+      if ($funderIndex) {
```

Comment:
Please check the documentation for the array_search function. It's not safe to use it like this.

Reply:
Since you use PhpStorm, it would be good if you could install the Php Inspections extension. It would have warned you about this